src/workflow/Logger.py

```diff
@@ -0,0 +1,29 @@
+from pathlib import Path
+import streamlit as st
+
+
+class Logger:
+    """
+    A simple logging class for writing messages to a log file. This class is designed
+    to append messages to a log file in the current workflow directory, facilitating
+    easy tracking of events, errors, or other significant occurrences in processes called
+    during workflow execution.
+
+    Attributes:
+        log_file (Path): The file path of the log file where messages will be written.
+    """
+    def __init__(self):
+        self.log_file = Path(st.session_state["workflow-dir"], "log.txt")
```

Comment:
better to pass a string to constructor at construction site instead of querying some global state here

Reply:
-> e.g. no dependency on the streamlit object here